### PR TITLE
Relax prohibition on string addition

### PR DESCRIPTION
**Pull request by jeffdparker**
Slack thread suggested that prohibition on string addition was excessive.  
Have relaxed prohibition, and moved discussion.

### CODE REVIEW

File: tracks/python/exercises/raindrops/mentoring.md [hunk 1]
```diff
@@ -31,30 +31,23 @@ def raindrops(val: int) -> str:
...
+    result = ''
     if number % 3 == 0:
-      factors = factors + 'Pling'
+      result = result + 'Pling'
     if number % 5 == 0:
-      factors = factors + 'Plang'
+      result = result + 'Plang'
     if number % 7 == 0:
-      factors = factors + 'Plong'
+      result = result + 'Plong'
```
Comment (cmccandless) [inline tracks/python/exercises/raindrops/mentoring.md]: It is more idiomatic to use the `__iadd__` operator.

```suggestion
      result += 'Plong'
```

File: tracks/python/exercises/raindrops/mentoring.md [hunk 2]
```diff
@@ -31,30 +31,23 @@ def raindrops(val: int) -> str:
...
 
 ```python
-def raindrops(number):
-    factors = ''
+def raindrops(number) 
+  
+    result = ''
     if number % 3 == 0:
-      factors = factors + 'Pling'
+      result = result + 'Pling'
```
Comment (cmccandless) [inline tracks/python/exercises/raindrops/mentoring.md]:
```suggestion
      result += 'Pling'
```

File: tracks/python/exercises/raindrops/mentoring.md [hunk 3]
```diff
@@ -31,30 +31,23 @@ def raindrops(val: int) -> str:
...
     if number % 5 == 0:
-      factors = factors + 'Plang'
+      result = result + 'Plang'
```
Comment (cmccandless) [inline tracks/python/exercises/raindrops/mentoring.md]:
```suggestion
      result += 'Plang'
```

**Comment by jeffdparker** [comment]
@cmccandless - I can't figure out how to edit the additions to replace + with --iadd--().  Perhaps I can do so after this has been merged.

**Comment by cmccandless** [comment]
@jeffdparker 
The `__iadd__()` operator is `+=`.

**Comment by jeffdparker** [comment]
Understood.  When I look at the file, I don't see any changes, and I didn't see a way to edit the current version

File: tracks/python/exercises/raindrops/mentoring.md (hunk 3 above)
Comment (cmccandless) [reply tracks/python/exercises/raindrops/mentoring.md]: @jeffdparker Just click "Commmit suggestion" above like the other two, and we'll be done here.

**Comment by SleeplessByte** [comment]
Thanks to you both 🚀

**Pull request merged**